react/bingo/src/bingo/GameLogic.js

```diff
@@ -0,0 +1,221 @@
+                    + challengeText + '. Better luck next time!');
+                this.cleanUp();
+            }
+    
+        }
+        this.setState({
+            failedLetters: failedLetters,
+            successLetters: successLetters,
+            challengeWord: challengeChars,
+            maskedWord: this.maskedChars,
+            bingoColor : bingoColor
+
+        }, () => console.log(this.state));
+
+    }
+
+    cleanUp = () => {
+        this.setState (
+            {
+                attemptLetter: '',
+                failedLetters: '',
+                successLetters: '',
+                challengeWord: [],
+                maskedWord: [],
+                isGameOver : true,
+                bingoColor: [
+                    'attempt-left',
+                    'attempt-left',
+                    'attempt-left',
+                    'attempt-left',
+                    'attempt-left'
+                ]
+            }
+        );        
+    }
+    gameOn = (challengeChars, challengeText) => {
+        let i = 0;
+        for (let chara of challengeChars) {
+            i++;
+            console.log(this.state.maskedWord);
+            if (this.state.attemptLetter === chara.toUpperCase()) {
+                let temp = [];
+                let j = 0;
+                for (let c of this.maskedChars) {
+                    j++;
+                    if (c === '*') {
+                        if (i === j) {
+                            temp.push(this.state.attemptLetter);
+                        } else {
+                            temp.push('*');
+                        }
+
+                    } else {
+                        temp.push(c);
+                    }
+                }
+                this.maskedChars = temp;
+                if (this.maskedChars.indexOf('*') === -1) {
+                    alert('Great Job !! you got ' + chara.toUpperCase() + ' at position ' + i);
+                    alert('Congratulations!!! You have guessed the word right. It is ' +
+                        challengeText);
+                    this.cleanUp();
+                }else {
+                    alert('Great Job !! you got ' + chara.toUpperCase() + ' at position ' + i);
+                }
+            }
+        }
+    }
+
+    render() {
+        return (
+            <div>
+                <h3>Guess a letter</h3>
+                <input
+                    className='guess-letter'
+                    maxLength='1'
+                    type="text"
+                    value={this.state.attemptLetter}
+                    onChange={(event) => this.setState({ attemptLetter: event.target.value.toUpperCase() })}></input>
+
+                <button className='button-style' onClick={() => this.handleSubmit()}> Enter </button>
+                <div>
+                    {this.getBlankText()}
+                </div>
+
+            </div>
+        );
+
+    }
+
+}
+
+class RenderChallenge extends React.Component {
```

Comment:
Can be in it's own file.

Reply:
Refactoring will be addressed in the new project. And the pure component was extended. Resolving